sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.controller.applicationChangeRequest.e2e-spec.ts

```diff
@@ -0,0 +1,307 @@
+import { HttpStatus, INestApplication } from "@nestjs/common";
+import { TestingModule } from "@nestjs/testing";
+import * as request from "supertest";
+import {
+  BEARER_AUTH_TYPE,
+  createTestingAppModule,
+  FakeStudentUsersTypes,
+  getStudentToken,
+  mockUserLoginInfo,
+} from "../../../testHelpers";
+import {
+  createE2EDataSources,
+  E2EDataSources,
+  saveFakeApplication,
+  getProviderInstanceForModule,
+  saveFakeStudent,
+  ensureProgramYearExists,
+} from "@sims/test-utils";
+import {
+  APPLICATION_EDIT_STATUS_IN_PROGRESS_VALUES,
+  ApplicationEditStatus,
+  ApplicationStatus,
+  ProgramYear,
+  RelationshipStatus,
+  StudentAssessmentStatus,
+} from "@sims/sims-db";
+import {
+  APPLICATION_CHANGE_REQUEST_ALREADY_IN_PROGRESS,
+  FormNames,
+  FormService,
+} from "../../../services";
+import { AppStudentsModule } from "../../../app.students.module";
+import { SaveApplicationAPIInDTO } from "../models/application.dto";
+import MockDate from "mockdate";
+
+describe("ApplicationStudentsController(e2e)-applicationChangeRequest", () => {
+  let app: INestApplication;
+  let appModule: TestingModule;
+  let db: E2EDataSources;
+  let programYear: ProgramYear;
+  let defaultPayload: SaveApplicationAPIInDTO;
+
+  beforeAll(async () => {
+    const { nestApplication, module, dataSource } =
+      await createTestingAppModule();
+    app = nestApplication;
+    appModule = module;
+    db = createE2EDataSources(dataSource);
+    // Program Year for the following tests.
+    const formService = await getProviderInstanceForModule(
+      appModule,
+      AppStudentsModule,
+      FormService,
+    );
+    programYear = await ensureProgramYearExists(db, 2050);
```

Comment:
Is there any particular reason to go for a program year way ahead in future?

Reply:
To avoid using the existing ones. It will work with any PY 2025 on, so instead of using 2025, which will potentially be deactivated a few years from now, I just used one in the far future.